### PR TITLE
internals improvements

### PR DESCRIPTION
**Pull request by vtjnash**
Some various improvements to some internals, observed while doing other work

### CODE REVIEW

File: src/codegen.cpp [hunk 1]
```diff
@@ -1777,7 +1777,7 @@ static std::pair<bool, bool> uses_specsig(jl_method_instance_t *lam, jl_value_t
     // not invalid, consider if specialized signature is worthwhile
     if (prefer_specsig)
         return std::make_pair(true, false);
-    if (!deserves_retbox(rettype) && !jl_is_datatype_singleton((jl_datatype_t*)rettype))
+    if (!deserves_retbox(rettype) && !jl_is_datatype_singleton((jl_datatype_t*)rettype) && rettype != (jl_value_t*)jl_bool_type)
```
Comment (aviatesk) [inline src/codegen.cpp:1780]: What's the rationale of this change ?

Comment (vtjnash) [reply src/codegen.cpp:1780]: This part estimates the calling convention cost, and there is no cost to a bool
